rs/node_rewards/canister/tests/regisry_sync_test.rs

```diff
@@ -0,0 +1,63 @@
+use candid::{Decode, Encode};
+use ic_interfaces_registry::RegistryValue;
+use ic_nns_test_utils::common::{build_node_rewards_test_wasm, NnsInitPayloadsBuilder};
+use ic_nns_test_utils::state_test_helpers::{
+    query, setup_nns_canisters, state_machine_builder_for_nns_tests,
+};
+use ic_node_rewards_canister_api::lifecycle_args::InitArgs;
+use ic_protobuf::registry::subnet::v1::{SubnetRecord, SubnetType};
+use ic_registry_keys::make_subnet_record_key;
+use ic_types::{PrincipalId, SubnetId};
+use std::time::Duration;
+
+#[test]
+fn test_registry_value_syncing() {
+    let state_machine = state_machine_builder_for_nns_tests().build();
```

Comment:
This is not a request for changes but a question so feel free to close:
Aren't StateMachine-based tests deprecated in favour of PocketIC-based tests? CC @mraszyk.

Reply:
I'd encourage test authors to convert their StateMachine tests to PocketIC, but I acknowledge that the conversion can only be done incrementally and it might be infeasible to convert this test to PocketIC as of now.

Reply:
I think when we last discussed it, we talked about a thin wrapper around PocketIc with the StateMachine interface for easier transition.

Part of the problem we're having is we have switched "preferred testing approaches" 3 times.  In the first case, we were using the canister_test crate.  Then we discovered StateMachine was a great tool (much faster, less flakier).  Then PocketIC came along.

Unfortunately PocketIC seems to be slower in CI (it uses more threads?  It sometimes times out).

That being said, I will look into writing it with different tools to make switching easier.

Reply:
Going to do that work in a followup PR to make it simpler